CMakeLists.txt

```diff
@@ -962,6 +962,10 @@ if (HIDAPI_FOUND OR LibUSB_COMPILE_TEST_PASSED)
   endif()
 endif()
 
+if(CMAKE_BUILD_TYPE STREQUAL "Debug")
+  set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -D LOKI_DEBUG")
```

Comment:
Defines in cmake should generally go through `add_definitions(-DLOKI_DEBUG)` rather than putting it into CXX flags.  (Though CMakeLists.txt has a hodgepodge of doing it both ways, we should try to stick to the more correct way for new additions).  I think also the define should be a bit more specific: perhaps `-DLOKI_DEBUG_COMMANDS`.


Separately, using the CMAKE_BUILD_TYPE to enable/disable this is a little weird -- the build type is really only meant to control the build optimization levels but this flag is actually changing the way the code operates.  This would be cleaner with a separate cmake option with something a bit more specific than just a generic "DEBUG" define, for example:

```
option(LOKI_DEBUG_COMMANDS "Enable lokid debugging commands" OFF)
if (LOKI_DEBUG_COMMANDS)
  add_definitions(-DLOKI_DEBUG_COMMANDS)
endif()
```

and then the Makefile can add a `-DLOKI_DEBUG_COMMANDS=ON` for the debug build, but a more fine-grained build that invokes CMake directly (which is how I build 99% of the time) can see and set the option with the description.  Another advantage is that it then becomes possible to do a release build but still have debug commands available for testing that things work properly on a release build (which can matter -- lokinet had a bug a while ago where there was an assert with a required side effect that made debug builds work but release builds not work).